block_cascade/cli/crud.py

```diff
@@ -0,0 +1,259 @@
+        "max_replica_count": CPU_MAX_REPLICA_COUNT,
+    }
+
+    CPU_POOL = {
+        "machine_spec": {
+            "machine_type": CPU_MACHINE_TYPE,
+        },
+        "replica_count": CPU_REPLICA_COUNT,
+        "disk_spec": DISK_SPEC,
+    }
+    if CPU_MIN_REPLICA_COUNT > 0 or CPU_MAX_REPLICA_COUNT > 0:
+        CPU_POOL["autoscaling_spec"] = CPU_AUTOSCALING_SPEC
+
+    GPU_POOL = {
+        "machine_spec": {
+            "machine_type": GPU_MACHINE_TYPE,
+            "accelerator_type": GPU_ACCELERATOR_TYPE,
+            "accelerator_count": GPU_ACCELERATOR_COUNT,
+        },
+        "replica_count": GPU_REPLICA_COUNT,
+        "disk_spec": DISK_SPEC,
+    }
+
+    if CPU_REPLICA_COUNT > 0 and GPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [CPU_POOL, GPU_POOL]
+    elif CPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [CPU_POOL]
+    elif GPU_REPLICA_COUNT > 0:
+        RESOURCE_POOLS = [GPU_POOL]
+    else:
+        RESOURCE_POOLS = []
+
+    PERSISTENT_RESOURCE = {
+        "display_name": resource.persistent_resource_id,
+        "resource_pools": RESOURCE_POOLS,
+    }
+
+    return PERSISTENT_RESOURCE
+
+
+def get_parent_str(resource: GcpResource) -> str:
+    """Return the parent string for the given project and region."""
+    project_id = resource.environment.project
+    region = resource.environment.region
+    return f"projects/{project_id}/locations/{region}"
+
+
+def get_endpoint_str(region: str) -> str:
+    """Return the parent string for the given project and region."""
+    return f"{region}-{SERVICE}"
+
+
+@click.command()
+@click.option(
+    "--config-name",
+    "-c",
+    required=True,
+    help="Name of the configuration block in cascade.yml to use.",
+)
+def create_persistent_resource(config_name: str):
+    """
+    Create a persistent resource from a configuration block in cascade.yaml.
+    Configuration block will be keyed on the config-name argument.
+    """
+    resource = find_default_configuration()[config_name]
+    persistent_resource = get_persistent_resource_payload(resource)
+
+    # CreatePersistentResoure SDK
+    region = resource.environment.region
+    client_options = {"api_endpoint": f"{region}-{SERVICE}"}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    parent = get_parent_str(resource)
+
+    try:
+        operation = client.create_persistent_resource(
+            parent=parent,
+            persistent_resource_id=resource.persistent_resource_id,
+            persistent_resource=persistent_resource,
+        )
+        click.echo(
+            """
+            Attempting to create Persistent Resource... 
+            This may take several minutes and will continue if you close this terminal.
+            """
+        )
+        response = operation.result()
+        response = json.loads(json_format.MessageToJson(response._pb))
+        click.echo(f'Persistent resource {response["name"]} created successfully.')
```

Comment:
Does this return when the persistent cluster is running for running jobs ? I just recall `gcloud` would return when scheduled the cluster creation (which took awhile) so wasn't sure if that behavior existed for the API as well (or it just blocks for a long time).

Reply:
Yes, it does eventually return, but takes a few minutes.